lib/media.js

```diff
@@ -279,26 +279,44 @@ module.exports = function(Cam) {
 	 */
 
 	/**
-	 * Get existing video encoder configuration options by its token
-	 * If token is omitted tries first from #videoEncoderConfigurations array
-	 * @param {string} [token] Token of the requested video encoder configuration
+	 * Get video encoder configuration options by video encoder configuration token or media profile token
+	 * If options is omitted, returns camera generic video encoder configuration options
+	 * If both token are set, returns video encoder configuration options compatible with both
+	 * If options is a string it is considered as configurationToken (for backward compatibility)
+	 * @param {object} [options]
+	 * @param {string} [options.configurationToken] The video encoder configuration token
+	 * @param {string} [options.profileToken] The media profile token
 	 * @param {Cam~VideoEncoderConfigurationOptionsCallback} callback
 	 */
-	Cam.prototype.getVideoEncoderConfigurationOptions = function(token, callback) {
+	Cam.prototype.getVideoEncoderConfigurationOptions = function(options, callback) {
 		if (callback === undefined) {
-			callback = token;
-			if (this.videoEncoderConfigurations && this.videoEncoderConfigurations[0]) {
-				token = this.videoEncoderConfigurations[0].$.token;
-			} else {
-				return callback(new Error('No video encoder configuration token is present!'));
-			}
+			callback = options;
```

Comment:
```suggestion
			callback = options;
			if (this.videoEncoderConfigurations && this.videoEncoderConfigurations[0]) {
				options = { configurationToken: this.videoEncoderConfigurations[0].$.token };
			} else {
				return callback(new Error('No video encoder configuration token is present!'));
			}
```

to avoid to break backwards compatibility